Run inside docker and ssh to another server

johnbradley opened the pull request:
Splits code into two parts.
One that contains a web server that runs inside docker. 
The other is to allow the docker container to ssh onto a server and create the flag file. This flag file will be detected by a root cron job that will restart a service and delete the flag file.

johnbradley committed:
run server in docker

johnbradley committed:
Update README.md

johnbradley committed:
Update README.md

johnbradley committed:
Update README.md

johnbradley committed:
TODO in authorized_keys

johnbradley committed:
Merge branch 'docker' of https://github.com/Duke-GCB/ElucidatorRestart into docker

johnbradley committed:
Update README.md

johnbradley committed:
Update README.md

johnbradley committed:
Update README.md

johnbradley committed:
at startup chown apache ssh directory

johnbradley committed:
Merge branch 'docker' of https://github.com/Duke-GCB/ElucidatorRestart into docker

johnbradley committed:
at startup chown apache ssh directory recursively

dleehr commented on README.md:
```diff
@@ -8,97 +9,45 @@ __User Clicks "Submit Restart Request"__
+     sp-key.pem                # generated by /etc/shibboleth/keygen.sh
+  ssh/
+     id_rsa                    # ssh setup used to connect to machine that will restart service
+     id_rsa.pub
```
The client doesn't need the public key - but if the keypair is generated on this host, it will naturally be there.

dleehr commented on var/www/cgi-bin/elucidator-restart.sh:
```diff
@@ -36,7 +33,7 @@ function GetResponse {
        <h1>Elucidator Restart Log</h1>
        <pre>"
-cat $RESTARTLOG
+ssh $ELUCIDATOR_HOST viewlog
```
Did you consider `ssh -i /path/to/id_rsa ...` instead of relying on the key placement in ~/.ssh? I suppose both are similarly awkward, but mounting apache's .ssh directory seems a little more convoluted.

johnbradley committed:
update domain name

johnbradley committed:
update shibbolet entity

johnbradley committed:
using alternative install mirror

johnbradley committed:
simplify setup by using full path to private key

johnbradley committed:
use pub key path everywhere

johnbradley committed:
remove chown for ssh dir

johnbradley committed:
known_hosts and timeout

use known hosts from external to simplify setup
set a timeout since the target server occasionally hangs

johnbradley committed:
fix typo in ssh command

johnbradley committed:
add timeout message
